signer/src/config/mod.rs

```diff
@@ -833,6 +667,55 @@ mod tests {
         ))
     }
 
+    #[test]
+    fn p2p_ip4_uri_works() {
+        use libp2p::multiaddr::Protocol;
+
+        clear_env();
+
+        std::env::set_var("SIGNER_SIGNER__P2P__LISTEN_ON", "tcp://0.0.0.0:4122");
+        let settings = Settings::new_from_default_config().expect("failed to load default config");
+
+        let actual = settings
+            .signer
+            .p2p
+            .listen_on
+            .first()
+            .expect("listen_on is empty");
+        let expected = Multiaddr::empty()
+            .with(Protocol::Ip4(
+                "0.0.0.0".parse().expect("failed to parse ipaddr"),
+            ))
+            .with(Protocol::Tcp(4122));
+
+        assert_eq!(*actual, expected);
+    }
+
+    #[test]
+    #[ignore] // TODO(513): Problem parsing ipv6 addresses
+    fn p2p_ip6_uri_works() {
+        use libp2p::multiaddr::Protocol;
+
+        clear_env();
+
+        std::env::set_var("SIGNER_SIGNER__P2P__LISTEN_ON", "tcp://ff06::c3:4122");
```

Comment:
Isn't the convention of using the square brackets necessary here? I mean like `tcp://[ff06::c3]:4122`, I assumed that without it that it wouldn't work?

Reply:
Ah, yes it is, I forgot to revisit this test, will do that before I merge and get rid of the skip.

Reply:
Fixed up in [66b5bde](https://github.com/stacks-network/sbtc/pull/521/commits/66b5bdef3bf2894281c852b300817ba671f927e7)